### PR TITLE
[firefly-iii] add guide for firefly iii

### LINKED ISSUE #626: Guide for firefly-iii
Got firefly-iii (https://www.firefly-iii.org/) running on my uberspace and now I'll reproduce the steps and write a guide.

### PR DESCRIPTION
**Pull request by FelixKohlen**
This is a guide to install Firefly III on uberspace. Feedback is appreciated, as this is my first time contributing to this project.

Fixes #626

### CODE REVIEW

**Comment by Systemsemaphore** [comment]
I was almost done with my guide for firefly-iii, was planning on uploading this weekend. Welp, guess I can delete that file now 😅 
Awesome work nonetheless! Thank you! 👍

**Comment by luto** [comment]
Thanks for you submission! I'm sorry, but I won't find time to review it until mid-January :/

File: source/guide_firefly-iii.rst [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+ [isabell@stardust ~]$ mysql -e "CREATE DATABASE ${USER}_fireflyiii"
+ [isabell@stardust ~]$
+
+``cd`` into your Firefly III directory.
+
+.. code-block:: console
+
+ [isabell@stardust isabell]$ cd firefly_iii
+ [isabell@stardust ~]$
```
Comment (luto) [inline source/guide_firefly-iii.rst:84]: the new working directory changes the prompt
```suggestion
 [isabell@stardust firefly_iii]$
```

File: source/guide_firefly-iii.rst [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+======================
+To finish the installation and setup your database run the following commands:
+
+.. code-block:: console
+ :emphasize-lines: 1,2,3
+
+ [isabell@stardust ~]$ php artisan migrate:refresh --seed
+ [isabell@stardust ~]$ php artisan firefly-iii:upgrade-database
+ [isabell@stardust ~]$ php artisan passport:install
+ [isabell@stardust ~]$
```
Comment (luto) [inline source/guide_firefly-iii.rst:100]: the new working directory changes the prompt :)

```suggestion
 [isabell@stardust firefly_iii]$ php artisan migrate:refresh --seed
 [isabell@stardust firefly_iii]$ php artisan firefly-iii:upgrade-database
 [isabell@stardust firefly_iii]$ php artisan passport:install
 [isabell@stardust firefly_iii]$
```

File: source/guide_firefly-iii.rst [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+.. warning:: Please make sure to backup your data before updating.
+
+To update your Firefly III installation, ``cd`` to the directory one level above your :manual:`document root <web-documentroot>`, then download the new version via composer. Replace <next_version> with the latest version.
+
+.. code-block:: console
+ :emphasize-lines: 1,2
+
+ [isabell@stardust ~]$ cd /var/www/virtual/$USER/
+ [isabell@stardust ~]$ composer create-project grumpydictator/firefly-iii --no-dev --prefer-dist firefly-iii-updated <next_version>
```
Comment (luto) [inline source/guide_firefly-iii.rst:147]: the new working directory changes the prompt :)

```suggestion
 [isabell@stardust isabell]$ composer create-project grumpydictator/firefly-iii --no-dev --prefer-dist firefly-iii-updated <next_version>
```

File: source/guide_firefly-iii.rst [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+ [isabell@stardust ~]$ cp firefly_iii/storage/export/* firefly-iii-updated/storage/export/
+ [isabell@stardust isabell]$
+
+Navigate into the newly created folder and run the following commands to finish the upgrade:
+
+.. code-block:: console
+ :emphasize-lines: 1
+
+ [isabell@stardust ~]$ cd firefly-iii-updated
+ [isabell@stardust ~]$ rm -rf bootstrap/cache/*
```
Comment (luto) [inline source/guide_firefly-iii.rst:168]: working directory, like above.

File: source/guide_firefly-iii.rst [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+ [isabell@stardust ~]$ php artisan cache:clear
+ [isabell@stardust isabell]$
+
+Move one folder up again and rename the updated folder, so our :manual:`document root <web-documentroot>` points to the updated version.
+
+.. code-block:: console
+ :emphasize-lines: 1,2,3
+
+ [isabell@stardust ~]$ cd ..
+ [isabell@stardust ~]$ mv firefly_iii firefly-iii-old
```
Comment (luto) [inline source/guide_firefly-iii.rst:182]: working directory, like above.

File: source/guide_firefly-iii.rst [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+
+ MAIL_MAILER=smtp
+ MAIL_HOST=stardust.uberspace.de
+ MAIL_PORT=587
+ MAIL_FROM=isabell@uber.space
+ MAIL_USERNAME=isabell@uber.space
+ MAIL_PASSWORD=MySuperSecretPassword
+ MAIL_ENCRYPTION=tls
+
+``MAIL_USERNAME`` and ``MAIL_PASSWORD`` must be the login data from a valid mail account.
```
Comment (luto) [inline source/guide_firefly-iii.rst:123]: Please include steps on how to create the required account.

**Review by luto — CHANGES_REQUESTED** [review]
Thanks for the guide! There seems to be a bit of confusion about the working directory in the prompt. Can you have another look, please? Once that is resolved, I'll take your guide for a spin :)

**Comment by FelixKohlen** [comment]
Sorry for the mess with the home directory, I completely forgot to change it while copy&pasting. Have a look at my changes, I think all your remarks should be addressed.

**Comment by FelixKohlen** [comment]
@madmooose Did you install it on a fresh uberspace and did you follow the steps to change the DocumentRoot? I tested the guide again on a clean installation to make all the changes luto requested and I did not have to change the .htaccess file on that run.
I know I had .htaccess problems on my first try when I did not symlink the DocumentRoot to the firefly/public directory.

**Comment by madmooose** [comment]
This could be the difference. I did the setup on my existing uberspace and used a subdomain. just saw that you need the rewritebase rule just for different documentroot (see here: https://manual.uberspace.de/web-documentroot/) . Maybe it's a good idea to mention it anyway here again. Just in case somebodys doing like i do (what of course is not recommended :) )

**Comment by luto** [comment]
thank you! :tada:

**Comment by luto** [comment]
@FelixKohlen I can't find your mail address. Please ping hallo@uberspace.de about your reward :)

**Pull request merged**